libc/vdso/vdso.c

```diff
@@ -3,22 +3,25 @@
 #include <sys/time.h>
 
 #ifdef __x86_64__
+time_t __vdso_kernel_time(time_t *tloc);
 __attribute__((__visibility__("default")))
 time_t __vdso_time(time_t *tloc)
 {
-    return time(tloc);
+    return __vdso_kernel_time(tloc);
```

Comment:
Won't it be clearer to just add the `arch::tls_switch_on_app_stack` here? What's the benefit of having another source file, another set of functions? Also the extra call isn't free.